akka-stream/src/main/scala/akka/stream/impl/Stages.scala

```diff
@@ -92,108 +93,122 @@ private[stream] object Stages {
 
   import DefaultAttributes._
 
+  // FIXME: To be deprecated as soon as stream-of-stream operations are stages
   sealed trait StageModule extends FlowModule[Any, Any, Any] {
-    def attributes: Attributes
     def withAttributes(attributes: Attributes): StageModule
     override def carbonCopy: Module = withAttributes(attributes)
   }
 
-  final case class StageFactory(mkStage: () ⇒ Stage[_, _], attributes: Attributes = stageFactory) extends StageModule {
-    override def withAttributes(attributes: Attributes) = copy(attributes = attributes)
+  /*
+   * Stage that is backed by a GraphStage but can be symbolically introspected
+   */
+  case class SymbolicGraphStage[-In, +Out, Ext](symbolicStage: SymbolicStage[In, Out])
+    extends PushPullGraphStage[In, Out, Ext](
+      symbolicStage.create,
+      symbolicStage.attributes) {
   }
 
-  final case class MaterializingStageFactory(
-    mkStageAndMaterialized: () ⇒ (Stage[_, _], Any),
-    attributes: Attributes = stageFactory) extends StageModule {
-    override def withAttributes(attributes: Attributes) = copy(attributes = attributes)
+  val identityGraph = SymbolicGraphStage[Any, Any, Any](Identity)
```

Comment:
this should be private (with underscore) plus a public getter that casts it for type T